client.go

```diff
@@ -63,6 +64,14 @@ func (c *Client) DirectionsMatrix(ctx context.Context, req *DirectionsMatrixRequ
 	return directionsMatrix(ctx, c, req)
 }
 
+func (c *Client) ReverseGeocode(ctx context.Context, req *ReverseGeocodeRequest) (*ReverseGeocodeResponse, error) {
+	return reverseGeocode(ctx, c, req)
```

Comment:
I missed this on the directions PR (or you already explained it to me) - Why not just keep keep c as the receiver and use `c.reverseGeocode(ctx, req)`?

Seems weird all this method is doing is switching it into a function?

Reply:
[haha, you're coming full circle on this](https://github.com/AirspaceTechnologies/go-mapbox/pull/1/commits/483fe280418f0f59b439728b92727aca412b8cc4#r357853943)! :D

My thoughts: I agree, something about it seems amiss. Why not do the following: 

- in `geocoding.go` -- rename `func reverseGeocode(...` to `func (c *Client) ReverseGeocode(...`
- in `client.go` delete the method there since it's already defined in `geocoding.go`

This is also what google does for some of their web apis ([example here](https://github.com/googlemaps/google-maps-services-go/blob/master/geocoding.go#L35))

Reply:
discussed over slack. decision is to stay as is (for now)